scapy/layers/dhcp6.py

```diff
@@ -413,8 +414,8 @@ class DHCP6OptIA_TA(_DHCP6OptGuessPayload):         # RFC sect 22.5
                    FieldLenField("optlen", None, length_of="iataopts",
                                  fmt="!H", adjust=lambda pkt, x: x + 4),
                    XIntField("iaid", None),
-                   _IATAOptField("iataopts", [], DHCP6OptIAAddress,
-                                 length_from=lambda pkt: pkt.optlen - 4)]
+                   PacketListField("iataopts", [], guess_payload_class_and_make_instance,
```

Comment:
Instead of this, you could set `_DHCP6OptGuessPayload` and add a `dispatch_hook` handler to the latter. That would be super clean.

See https://stackoverflow.com/a/28025620/5459467 for more infos